modules/core/src/main/java/org/apache/ignite/internal/processors/metric/impl/MetricUtils.java

```diff
@@ -163,4 +168,69 @@ private static boolean ensureAllNamesNotEmpty(String... names) {
 
         return names;
     }
+
+    /**
+     * Abbreviates package name for metric naming purposes.
+     *
+     * @param pkgNameDepth Exhibition level of java package name. The bigger, the wider.
+     *                     Max level is {@link #MAX_ABBREVIATE_NAME_LVL}. Values:
+     *                     <pre>
+     *                         0 - wont add package name;
+     *                         1 - add only first char of each name in java package;
+     *                         2 - add first and last char of each name in java package;
+     *                         Any other - add full name of java package.
+     *                     </pre>
+     * @return Abbreviated name of {@code cls}.
+     */
+    public static String abbreviateName(Class<?> cls, int pkgNameDepth) {
```

Comment:
@NSAmelchev , in your example you two different services and their metric name has different service name like different domain. However, in particular case it would require some efforts to distinguish metrics. If methods like in your case are contained in same interface. I think it is rare case and should not affect common simplicity of metric names. I suggest to keep metric names as simple as possible. Available improvement is adding 2 chars of package name:
void_testMethod(Param)
void_testMethod(og.ae.ie.il.ps.se.ir.il.Param)

Still possible to get abbreviated package name complicated when firs/last chars are the same. I believe keeping rape case confusing is better that keep long names for 99% cases. WDYT?